sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java

```diff
@@ -6,7 +6,7 @@
  * Utility methods for checking whether version strings are valid SLS version strings.
  */
 public class SlsProductVersions {
-    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-[a-z0-9-.]*$");
+    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+((-[a-z0-9-.]*)|(\\.dirty))$");
```

Comment:
ah, I missed that. I think I'd prefer to remove `.` support -- we added it just to support `.dirty` anyway.